### PR TITLE
Redux changes

### PR DESCRIPTION
**Pull request by shawwwwwwww**
@AbhinavShah421 let me know how we're doing with this. creating a draft PR to track.

### CODE REVIEW

**Comment by AbhinavShah421** [comment]
@shawwwwwwww  I want Gehric to review these changes first, So I can make changes if needed

File: packages/client-core/src/user/store/UserState.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ const state = createState({
 
 export const userReducer = (_, action: UserActionType) => {
   Promise.resolve().then(() => userReceptor(action))
-  return state.attach(Downgraded).value
+  return state
```
Comment (speigg) [inline packages/client-core/src/user/store/UserState.ts:19]: Please leave this the way it was. Need to export "downgraded" values (values that aren't wrapped in hookstate proxy objects).

File: packages/client-core/src/media/components/video/store/VideoState.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import { VideoActionType } from './VideoAction'
+
+const state = createState({
+  videos: [] as Array<PublicVideo>,
+  error: ''
+})
+
+export const VideoReducer = (_, action: VideoActionType) => {
+  Promise.resolve().then(() => VideoReceptor(action))
+  return state
```
Comment (speigg) [inline packages/client-core/src/media/components/video/store/VideoState.ts:12]: Need to export "downgraded" value here. See UserState.ts

File: packages/client-core/src/world/store/SceneState.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+const state = createState({
+  scenes: [] as Array<Scene>,
+  currentScene: null,
+  error: ''
+})
+
+export const SceneReducer = (_, action: SceneActionType) => {
+  Promise.resolve().then(() => SceneReceptor(action))
+  return state
```
Comment (speigg) [inline packages/client-core/src/world/store/SceneState.ts:13]: Need to export downgraded value here (see UserState.ts)

File: packages/docs/sidebars.js [hunk 4]
```diff
@@ -16,14 +16,14 @@ module.exports = {
...
-              'gameserver': require('./sidebar/typedoc-gameserver.js'),
-              'server-core': require('./sidebar/typedoc-server-core.js'),
+              // 'server': require('./sidebar/typedoc-server.js'),
+              // 'common': require('./sidebar/typedoc-common.js'),
+              // 'engine': require('./sidebar/typedoc-engine.js'),
+              // 'gameserver': require('./sidebar/typedoc-gameserver.js'),
+              // 'server-core': require('./sidebar/typedoc-server-core.js'),
             }
           ],
         },
```
Comment (speigg) [inline packages/docs/sidebars.js:29]: Are these changes related? It looks like they belong to a different PR

File: packages/engine/src/scene/behaviors/createPortal.ts [hunk 5]
```diff
@@ -21,6 +21,7 @@ export type PortalProps = {
   spawnPosition: Vector3
   spawnRotation: Quaternion
   spawnEuler: Euler
+  location: string
 }
 
 const vec3 = new Vector3()
```
Comment (speigg) [inline packages/engine/src/scene/behaviors/createPortal.ts:27]: Unrelated?

**Review by speigg — CHANGES_REQUESTED** [review]
I don't think we need these:

common/src/interfaces/Scene.ts 
packages/common/src/interfaces/Video.ts

I left several comments. The biggest thing missing is actually using the new hooked states properly in React components; need to remove them from Props, and just use them via `useVideoState()` or `useSceneState()` directly inside the Components that need them. Please see how `useUserState()` is used for reference. 

Overall, looking good, thanks!

File: packages/docs/sidebars.js (hunk 4 above)
Comment (AbhinavShah421) [reply packages/docs/sidebars.js:29]: Its unrelated to this, I'll revert it back.

File: packages/engine/src/scene/behaviors/createPortal.ts (hunk 5 above)
Comment (AbhinavShah421) [reply packages/engine/src/scene/behaviors/createPortal.ts:27]: Its unrelated to this, I'll revert it back.

File: packages/client-ml/tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,6 @@
+{
+    "extends": "../../tsconfig.json",
+    "compilerOptions": {
+      "outDir": "./lib"
+    },
+  }
```
[the remark this answers is not in the document]
Comment (AbhinavShah421) [reply packages/client-ml/tsconfig.json:6]: Its unrelated to this, I'll revert it back.

**Comment by HexaField** [comment]
Github is reporting many files as changing permissions. Please revert this. See https://stackoverflow.com/questions/28268842/changing-file-mode-permissions-from-100755-to-100644-on-github

**Comment by AbhinavShah421** [comment]
> Github is reporting many files as changing permissions. Please revert this. See https://stackoverflow.com/questions/28268842/changing-file-mode-permissions-from-100755-to-100644-on-github

Ok I am fixing this
